nncf/experimental/common/pruning/propagation_data.py

```diff
@@ -11,24 +11,54 @@
  limitations under the License.
 """
 
+from dataclasses import dataclass
 import json
-from typing import Dict
+from typing import Dict, Optional, Set
 from typing import List
 
+@dataclass
+class ProducerInfo:
+    """
+    Defines a node that produce pruning. Operation that corresponds to the node has built-in weights.
+    Pruning dimension is required in comparison with consumer info.
 
-class MaskProducer:
+    :param node_id: id of the node that produce pruning
+    :param pruning_dimension: axis number from 0 to N-1 in weights along which the pruning structure is defined.
+        N is total number of dimensions.
     """
-    Defines producer of the pruning.
+    node_id: int
+    pruning_dimension: int = 0
+
+    def __hash__(self) -> int:
+        return hash((self.node_id, self.pruning_dimension))
+
+    def __str__(self) -> str:
+        return str(self.node_id)
+
+@dataclass
+class ConsumerInfo:
+    """
+    Defines the node that absorbs (consume) pruning. There are 2 types of consumer. First, have built-in weights,
+    therefore can consume and produce pruning along the input channels. Consumers of the second type don't have built-in
+    weights, instead weights are coming as a second input. In that case, operation can only consume pruning and
+    pruning dimension is None.
+
+    :param node_id: id of the node that consume pruning.
+    :param pruning_dimension: axis number from 0 to N-1 in weights along which the pruning structure is defined.
+        N is total number of dimensions. Equals None for the second type of consumers.
```

Comment:
```suggestion
    Defines the node that absorbs (consume) pruning masks. There are 2 types of consumer. First, have built-in weights,
    therefore can consume and produce pruning along the input channels. Consumers of the second type don't have built-in
    weights, instead weights are coming as a second input. In that case
    pruning dimension is None.

    :param node_id: id of the node that consume pruning.
    :param pruning_dimension: axis number from 0 to N-1 in weights along which the pruning structure is defined.
        N is total number of dimensions. Equals None for the second type of consumers.
```
Second type of consumers could be possibly pruned, maybe in future we will support it! I suggest to drop the line `In that case, operation can only consume pruning`

Reply:
Done